src/DotNetWorker.Core/FunctionsApplication.cs

```diff
@@ -17,28 +19,52 @@ internal partial class FunctionsApplication : IFunctionsApplication
         private readonly ConcurrentDictionary<string, FunctionDefinition> _functionMap = new ConcurrentDictionary<string, FunctionDefinition>();
         private readonly FunctionExecutionDelegate _functionExecutionDelegate;
         private readonly IFunctionContextFactory _functionContextFactory;
+        private readonly IFunctionInvocationManager _functionInvocationManager;
         private readonly IOptions<WorkerOptions> _workerOptions;
         private readonly ILogger<FunctionsApplication> _logger;
         private readonly IWorkerDiagnostics _diagnostics;
 
-        public FunctionsApplication(FunctionExecutionDelegate functionExecutionDelegate, IFunctionContextFactory functionContextFactory,
-             IOptions<WorkerOptions> workerOptions, ILogger<FunctionsApplication> logger, IWorkerDiagnostics diagnostics)
+        public FunctionsApplication(
+            FunctionExecutionDelegate functionExecutionDelegate,
+            IFunctionContextFactory functionContextFactory,
+            IFunctionInvocationManager functionInvocationManager,
+            IOptions<WorkerOptions> workerOptions,
+            ILogger<FunctionsApplication> logger,
+            IWorkerDiagnostics diagnostics)
         {
             _functionExecutionDelegate = functionExecutionDelegate ?? throw new ArgumentNullException(nameof(functionExecutionDelegate));
             _functionContextFactory = functionContextFactory ?? throw new ArgumentNullException(nameof(functionContextFactory));
+            _functionInvocationManager = functionInvocationManager ?? throw new ArgumentNullException(nameof(functionInvocationManager));
             _workerOptions = workerOptions ?? throw new ArgumentNullException(nameof(workerOptions));
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _diagnostics = diagnostics ?? throw new ArgumentNullException(nameof(diagnostics));
         }
 
         public FunctionContext CreateContext(IInvocationFeatures features)
         {
+            CancellationTokenSource? cancellationTokenSource = null;
             var invocation = features.Get<FunctionInvocation>() ?? throw new InvalidOperationException($"The {nameof(FunctionInvocation)} feature is required.");
 
             var functionDefinition = _functionMap[invocation.FunctionId];
             features.Set<FunctionDefinition>(functionDefinition);
 
-            return _functionContextFactory.Create(features);
+            if (functionDefinition.BindsToCancellationToken)
+            {
+                cancellationTokenSource = new CancellationTokenSource();
+            }
+
+            var token = cancellationTokenSource is null ? CancellationToken.None : cancellationTokenSource.Token;
+            var context = _functionContextFactory.Create(features, token);
+
+            var invocationDetails = new FunctionInvocationDetails()
```

Comment:
I believe that the context needs to have the token so that we can expose it as a parameter via the CancellationTokenConverter. FunctionInvocationDetails keeps track of the tsc that we use to cancel the cancellation token and we don't expose that to customers